src/sentry/workflow_engine/metric_helpers.py

```diff
@@ -0,0 +1,166 @@
+from sentry.incidents.grouptype import MetricAlertFire
+from sentry.incidents.models.alert_rule import (
+    AlertRule,
+    AlertRuleThresholdType,
+    AlertRuleTrigger,
+    AlertRuleTriggerAction,
+)
+from sentry.snuba.models import QuerySubscription
+from sentry.users.services.user import RpcUser
+from sentry.workflow_engine.models import (
+    Action,
+    AlertRuleDetector,
+    AlertRuleTriggerDataCondition,
+    AlertRuleWorkflow,
+    DataCondition,
+    DataConditionGroup,
+    DataConditionGroupAction,
+    DataSource,
+    Detector,
+    DetectorState,
+    DetectorWorkflow,
+    Workflow,
+    WorkflowDataConditionGroup,
+)
+from sentry.workflow_engine.models.data_condition import Condition, ConditionType
+from sentry.workflow_engine.types import ActionType, DataSourceType, DetectorPriorityLevel
+
+
+def create_metric_action(alert_rule_trigger_action: AlertRuleTriggerAction) -> None:
+    try:
+        alert_rule_trigger_data_condition = AlertRuleTriggerDataCondition.objects.get(
+            alert_rule_trigger=alert_rule_trigger_action.alert_rule_trigger
+        )
+    except AlertRuleTriggerDataCondition.DoesNotExist:
+        return None
```

Comment:
Trying to think through if raising or returning a bool for created or not would be more helpful here

Reply:
🤔 you might want to break down the method a little and return tuples of the initialized data. 

by doing that, the methods would be a bit more composable, and you could have many small methods and at the end have one that calls all sorts of stuff. maybe something along these lines: https://github.com/getsentry/sentry/blob/master/tests/sentry/workflow_engine/test_base.py#L65?

What I've liked about that code is that we can easily create the core set of models, then have other models add associations to those core ones later and build up to a larger more complex model. Here's an example of that design in practice; https://github.com/getsentry/sentry/pull/81975/files#diff-7939c037fe7efe66cff98b2df8b4234bcfbf5f2270f028ab7e5259a7322bf73dR20

----

for this exception, we should probs also add a log so we can see it when executing scripts or in production if this code is used in the notification action.